File: 9-palindrome-number/palindrome-number.cpp
class Solution {
public:
    bool isPalindrome(int x) {
        if(x<0) return false;
        int original = x;
        long rev_x=0;
        while(x!=0){
            rev_x=rev_x*10 + x%10;
            x=x/10;
        }return original==rev_x;
    }
};

File: 9-palindrome-number/palindrome-number.py
class Solution:
    def isPalindrome(self, x: int) -> bool:
        if x<0:
            return False;
        else:
            original =x;
            rev=0;
            while x != 0:
                d = x % 10
                rev = rev * 10 + d
                x = x // 10

            return original == rev